packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -337,3 +339,19 @@ export const getSuperClass = (
   );
   return undefined;
 };
+
+export const isConstructorFieldInitializer = (
```

Comment:
Slightly unfortunate that this doesn't require that the expression is a direct child of a constructor

Could we instead write this like:

```typescript
export const getConstructorFieldInitializers = *(
  member: ts.ConstructorDeclaration>, 
  typescript: typeof ts
): IterableIterator<ConstructorFieldInitializer> => { ... };
```

or the forEach style equivalent

Reply:
Sorry, I forgot to set the correct base branch for this PR. This change is actually in https://github.com/lit/lit/pull/4260 and even there it's only a refactor of existing logic into a function call for better readability. The function is already called only against direct child statements of the constructor body in practice, within a loop against those body children.

Another refactor that could accomplish your goal would be to pass the constructor body node in to verify that it's the parent, however I'm not 100% sure that that check is correct. I want to verify against TypeScript's field inference because things like `(this.foo = 1)` might count as a field even though they're in a nested expression.